pkg/client/dtclient/settings_client.go

```diff
@@ -436,7 +449,7 @@ func (d *SettingsClient) Upsert(ctx context.Context, obj SettingsObject, upsertO
 		}
 	}
 
-	payload, err := buildPostRequestPayload(ctx, obj, externalID, upsertOptions.InsertAfter)
+	payload, err := buildPostRequestPayload(ctx, remoteObjectId, obj, externalID, upsertOptions.InsertAfter)
```

Comment:
If it is not possible to have both a remoteObjectId and an externalID here, should we have two separate functions for building post requests? Alternatively we can examine this is the larger refactoring.